packages/matter-node.js/test/IntegrationTest.ts

```diff
@@ -1505,29 +1505,15 @@ describe("Integration Test", () => {
     });
 
     after(async () => {
-        const promise = mdnsBroadcaster.expireAllAnnouncements(matterPort);
-
-        await MockTime.yield();
-        await MockTime.yield();
-        await MockTime.yield();
-        await MockTime.yield();
-        await MockTime.advance(150);
-        await MockTime.advance(150);
-        await MockTime.yield();
-
-        await MockTime.yield();
-        await MockTime.yield();
-        await MockTime.yield();
-        await MockTime.yield();
-        await MockTime.advance(150);
-        await MockTime.advance(150);
-        await MockTime.yield();
-
-        await promise;
+        // For closing all down we need to use the real Time implementation
```

Comment:
Could we maybe do yield/advance in a loop until the expireAllAnnouncements promise resolves?  Could even do it as a utility on mock time.

Reply:
Oops didn't mean to approve but you get this one for free ;)

Reply:
innthe meantime it is not just expiring it is also more stuff happening under the hood, mainly because we add more automatisms ... no idea how we can handle that in general  I add a todo in next commit to clean that up, ok?

Reply:
Sure sounds good